core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,217 @@
+        if (cancelInitiated
+            || (pollResponse != null && pollResponse.getStatus() != PollResponse.OperationStatus.IN_PROGRESS)) {
+            return;
+        }
+        //Time to call cancel
+        cancelOperation.accept(this);
+        cancelInitiated = true;
+    }
+
+    /**
+     * This will poll and send PollResponse. If you had stopped polling erlier, we will enable polling again.
+     * @return Return poll response as Flux
+     */
+    public Flux<PollResponse<T>> poll() {
+        if (fluxDisposable != null && !fluxDisposable.isDisposed()) {
+            fluxDisposable.dispose();
+        }
+        return fluxHandle;
+    }
+
+    /**
+     * This will block till poll operation is complete
+     * @return returns poll response
+     */
+    public PollResponse<T> block() {
+        return poll().blockLast();
+    }
+
+    /*
+     * Get whether or not this PollStrategy's long running operation is done.
+     * @return Whether or not this PollStrategy's long running operation is done.
+     */
+    private Mono<PollResponse<T>> sendPollRequestWithDelay() {
+        return Mono.defer(() -> delayAsync().then(Mono.defer(() -> {
+            if (!isAutoPollingEnabled() && !isTerminalState()) {
+                pollResponse = pollOperation.apply(pollResponse);
+            } else if (!isTerminalState()) {
+                return Mono.empty();
+            }
+            return Mono.just(pollResponse);
+        })));
+    }
+
+    /*
+     * Helper function
+     */
+    private boolean isTerminalState() {
+        return pollResponse != null && (
+            pollResponse.getStatus() == PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED
+                || pollResponse.getStatus() == PollResponse.OperationStatus.FAILED
+                || pollResponse.getStatus() == PollResponse.OperationStatus.USER_CANCELLED);
+
+    }
+
+    /*
+     * If this PollerOptions has a pollIntervalInMillis value, return an Mono that is delayed by the
+     * number of seconds that are in the pollIntervalInMillis value. If this PollerOptions doesn't have
+     * a pollIntervalInMillis value, then return an Single with no delay.
+     *
+     * @return A Mono with delay if this PollerOptions has a pollIntervalInMillis value.
+     */
+    private Mono<Void> delayAsync() {
+        Mono<Void> result = Mono.empty();
+        if (this.pollerOptions.pollInterval().toNanos() > 0) {
+            result = result.delaySubscription(this.pollerOptions.pollInterval());
+        }
+        return result;
+    }
+
+    /**
+     * 
+     * @param  autoPollingEnabled  true : Ensures the polling is happening in background.
+     *                             false : Ensures that polling is not happening in background.
+     */
+    public void setAutoPollingEnabled(boolean autoPollingEnabled) {
+        this.autoPollingEnabled = autoPollingEnabled;
+        if (this.autoPollingEnabled) {
+            if (fluxDisposable == null || fluxDisposable.isDisposed()) {
```

Comment:
Should we add a quick helper for this check? It's getting used in a lot of places

Reply:
resolved